pkg/rtc/transport.go

```diff
@@ -876,18 +779,31 @@ func (t *PCTransport) AddNegotiationPending(publisherID livekit.ParticipantID) {
 
 func (t *PCTransport) Negotiate(force bool) {
 	if force {
+		t.lock.Lock()
 		t.debouncedNegotiate(func() {
 			// no op to cancel pending negotiation
 		})
-		if err := t.CreateAndSendOffer(nil); err != nil {
-			t.params.Logger.Errorw("could not negotiate", err)
-		}
-	} else {
-		t.debouncedNegotiate(func() {
-			if err := t.CreateAndSendOffer(nil); err != nil {
-				t.params.Logger.Errorw("could not negotiate", err)
-			}
+		t.debouncePending = false
```

Comment:
Introducing this variable as each unforced negotiate restarts the debounce and pushes things back by 150ms.